scopes/envs/envs/envs.docs.md

```diff
@@ -1,33 +1,797 @@
+
+export class CustomReactExtension {
+  constructor(private react: ReactMain) {}
+
+  // Set the necessary dependencies to be injected (by Bit) into the following 'provider' function
+  static dependencies: any = [EnvsAspect, ReactAspect];
+
+  static async provider([envs, react]: [EnvsMain, ReactMain]) {
+    // The 'compose' methods to compose the overrides into a single environment
+    const customReactEnv = react.compose([
+      // This is were the environment's 'transformers' will be used to customize it
+    ]);
+
+    // Register this extension as an environment using the "registerEnv" slot (provided by the Environments aspect).
+    envs.registerEnv(customReactEnv);
+
+    return new CustomReactExtension(react);
+  }
+}
+```
+
+```ts
+// index.ts
+
+import { CustomReactExtension } from './custom-react.extension';
+export { CustomReactExtension };
+export default CustomReactExtension;
+```
+
+#### 3. Track the extension component
+
+We'll then track the new component (with the 'my-extensions' namespace):
+
+```shell
+$ bbit add extensions/custom-react -n my-extensions
+```
+
+#### 4. Set the extension component in the workspace config file
+
+Our extension component now has a component ID that can be used in our `workspace.jsonc` configuration file:
+
+```json
+{
+  "teambit.workspace/workspace": {
+    "name": "my-workspace",
+    "icon": "https://image.flaticon.com/icons/svg/185/185034.svg",
+    "defaultScope": "my-org.my-extensions"
+  },
+  "teambit.workspace/variants": {
+    "*": {
+      "my-org.my-extensions/custom-react": {}
+    }
+  }
+}
+```
+
+### The anatomy of an environment extension
+
+An environment extension uses the following Bit components to extend an existing environment, and to register itself as an environment:
+
+- The **"base" environment** (e.g, `@teambit/react`) is extended and customized using its override methods. Each override method, or "environment transformer", corresponds to a Bit extension component used by the environment (e.g, the TypeScript component). Using an 'environment transformer' will add new configurations to the relevant Bit component and will override any conflicting ones.<br /> The full list of available 'environment transformers' can be seen in the specific environment's documentation (see: React, React Native, Node).
+
+- The **'Environments' component** (`@teambit/envs`) is used to:
+  1. Register the new environment using its [slot](TODO)
+  2. Override a ["service handler"](TODO). This is done to replace a Bit component used by an environment service. For example, to set the "compiler" service handler to use Babel instead of TypeScript (see an example, [here](/docs/environments/build-environment#override-a-service-handler)).
+
+#### Override the config for a Bit component used by the environment
+
+The example below is of a React environment extension. This new environment overrides React's DevServer configuration by setting a new Webpack configuration file.
```

Comment:
consider mention that those APIs are about to be deprecated / replaced soon

Reply:
Done.